Sources/Features/Client/BacktraceOomWatcher.swift

```diff
@@ -0,0 +1,220 @@
+            // ignore errors or use do/catch block to handle errors more gracefully
+            try? FileManager.default.removeItem(at: oomFilePath)
+        }
+    }
+    
+    internal static func getAppVersion() -> String {
+        if let appVersion = Bundle.main.object(forInfoDictionaryKey: kCFBundleVersionKey as String) as? String {
+            return appVersion
+        } else {
+            return ""
+        }
+    }
+    
+    public func start() {
+        sendPendingOomReports()
+        // override previous application state after reading all information
+        saveState()
+    }
+    
+    internal static func getStatusFilePath() -> URL? {
+        // oom status file is available in application cache dir - the same dir
+        // where we store attributes
+        guard let cacheDir = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        return cacheDir.appendingPathComponent("BacktraceOomState.plist")
+    }
+    
+    private func sendPendingOomReports() {
+        
+        // if oom state file doesn't exist it means that we deleted it to
+        // prevent sending false oom crashes
+        if let oomFilePath = BacktraceOomWatcher.oomFilePath {
+            if !FileManager.default.fileExists(atPath: oomFilePath.path) {
+                return
+            }
+        } else {
+            // path to status file is empty or unvailable
+            return
+        }
+        
+        guard let appState = self.loadPreviousState() else {
+            return
+        }
+        
+        if !shouldReportOom(appState: appState) {
+            return
+        }
+        
+        
+        reportOom(appState: appState)
+        BacktraceOomWatcher.clean()
+    }
+    
+    private func shouldReportOom(appState: ApplicationInfo) -> Bool {
+        // no low memory warning
+        if appState.resource == nil {
+            return false
+        }
+        
+        // check if debugger was enabled
+        if appState.debugger {
+            // detected debugger in previous session
+            return false
+        }
+        // check system update
+        if appState.version != ProcessInfo.processInfo.operatingSystemVersionString {
+            // detected system update
+            return false
+        }
+        
+        // check application update
+        
+        if appState.appVersion != BacktraceOomWatcher.getAppVersion() {
+            // detected app update
+            return false
+        }
+        return true
+    }
+    
+    private func reportOom(appState: ApplicationInfo) {
+        guard let reportData = appState.resource else {
+            return
+        }
+        
+        var reportAttributes: Attributes = [:]
+        if let stateAttributes = appState.attributes {
+            if let attributes = try? JSONSerialization.jsonObject(with: stateAttributes, options: []) as? Attributes {
```

Comment:
It can be a one-liner: 
```swift
if let stateAttributes = appState.attributes, let attributes = try? JSONSerialization.jsonObject(with: stateAttributes, options: []) as? Attributes {
```